pkg/service/apikey.go

```diff
@@ -105,6 +105,7 @@ func (s *apiKeyService) Get(ctx context.Context, req *rpcv3.ApiKeyRequest) (*mod
 	if err == sql.ErrNoRows {
 		return nil, nil
 	}
+	GenerateApiKeyAuditEvent(ctx, s.al, AuditActionGenerate, req.Username)
```

Comment:
am confused here, why are we auditing a generate api key event for Get and GetByKey methods? I don't think these are required. Is there any other reason for adding these that I am missing.

There is already an audit event entry as part of Create, type cast condition seems to be wrong which is the reason we are not logging those.

Reply:
Had discussed this with @meain , will check once again

Reply:
Yeah, don't think we should have this here. Instead let us create a `AuditActionDownload` event for apikey in `user.go:RetrieveCliConfig`.